client/gutenberg/extensions/tiled-gallery/view.js

```diff
@@ -1,6 +1,117 @@
 /** @format */
 
+/**
+ * External dependencies
+ */
+import { throttle } from 'lodash';
+import ResizeObserver from 'resize-observer-polyfill';
+
 /**
  * Internal dependencies
  */
+import { RESIZE_RATE_IN_MS, DEFAULT_GALLERY_WIDTH } from './constants';
+import { getActiveStyleName, getLayout } from './layouts';
+
+/**
+ * Styles
+ */
 import './view.scss';
+
+const applyNodeSize = ( node, { width, height } ) => {
+	node.style.width = `${ width }px`;
+	node.style.height = `${ height }px`;
+};
+
+/**
+ * Calculate new size for the gallery and apply it
+ */
+const resizeGallery = ( { galleryNode, width, columns, layout } ) => {
+	const tileCount = galleryNode.querySelectorAll( '.tiled-gallery__item' ).length;
+
+	const galleryLayout = getLayout( {
+		columns,
+		layout,
+		tileCount,
+		width,
+	} );
+
+	// Resize rows within the gallery
+	galleryNode.childNodes.forEach( ( rowNode, rowIndex ) => {
+		const rowLayout = galleryLayout[ rowIndex ];
+		applyNodeSize( rowNode, rowLayout );
+
+		// Resize tiles within the row
+		const tileNodes = rowNode.querySelectorAll( '.tiled-gallery__item' );
+		tileNodes.forEach( ( tileNode, tileIndex ) => {
+			const tileLayout = rowLayout.tiles[ tileIndex ];
+			applyNodeSize( tileNode, tileLayout );
+		} );
+	} );
+};
+
+/**
+ * Throttled resize event
+ * Compares gallery width to its previous width to decide if to resize it.
+ */
+const throttleOnResize = throttle( entries => {
```

Comment:
I'm wondering if we would get better performance by using `debounce` and how that would look visually when resizing the window.

Differences: https://css-tricks.com/debouncing-throttling-explained-examples/